### PR TITLE
duckdb: init at 0.1.7

### PR DESCRIPTION
**Pull request by costrouc**
###### Motivation for this change

I wanted an alternative to sqlite for a column store database. duckdb was on hackernews and made a compelling case for a column store sqlite database.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by costrouc** [comment]
@GrahamcOfBorg build duckdb python3Packages.duckdb python2Packages.duckdb

File: pkgs/development/python-modules/duckdb/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  '';
+
+  nativeBuildInputs = [
+    pybind11
+  ];
+
+  buildInputs = [
+    setuptools_scm
+    pytestrunner
+  ];
```
Comment (jonringer) [inline pkgs/development/python-modules/duckdb/default.nix:30]: these should be in nativebuildInputs

Comment (costrouc) [reply pkgs/development/python-modules/duckdb/default.nix]: resolved

**Comment by costrouc** [comment]
@GrahamcOfBorg build duckdb python3Packages.duckdb python2Packages.duckdb

File: pkgs/development/python-modules/duckdb/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+, setuptools_scm
+, pytestrunner
+, pytest
+}:
+
+buildPythonPackage rec {
+  pname = "duckdb";
+  version = duckdb.version;
+
+  src = duckdb.src;
```
Comment (jonringer) [inline pkgs/development/python-modules/duckdb/default.nix:16]: a little more concise, but I don't feel super strongly about it:
```suggestion
  inherit (duckdb) version src;
```

Comment (costrouc) [reply pkgs/development/python-modules/duckdb/default.nix:16]: I like that. Shows intent better.

Comment (costrouc) [reply pkgs/development/python-modules/duckdb/default.nix]: resolved.

**Comment by costrouc** [comment]
@GrahamcOfBorg build duckdb python3Packages.duckdb python2Packages.duckdb

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 89000 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>17 packages marked as broken and skipped:</summary>  - digitalbitbox  - iterm2  - linuxPackages_4_4.evdi  - linuxPackages_hardkernel_4_14.digimend  - linuxPackages_hardkernel_4_14.evdi  - linuxPackages_hardkernel_4_14.nvidia_x11_legacy390  - linuxPackages_hardkernel_4_14.nvidiabl  - muslCross  - newlibCross  - octave-jit  - php72Packages.phpmd  - php73Packages.phpmd  - php74Extensions.pcs  - php74Extensions.pthreads  - python37Packages.notify  - python37Packages.pyblock  - python38Packages.notify</details><details>  <summary>4 packages built:</summary>  - duckdb  - python27Packages.duckdb  - python37Packages.duckdb  - python38Packages.duckdb</details>

**Pull request merged**